controllers/pkg/reconcilers/bootstrap/README.md

```diff
@@ -0,0 +1,15 @@
+# bootstrap controller
+
+The bootstrap controller is a k8s controller which goal is to bootstrap packages on a newly installed cluster. This ensure e.g. that a gitops tool like `config-sync` is installed on a cluster and subsequent configurations tasks get handled through the gitops toolchain.
+
+## implementation
+
+The controller acts on a secret. This ensures that the bootstrap controller is agnostic to the cluster implementation. It figures out the type of cluster based on the signatures of the secret. (right now only cluster api is implemented, but the code is able to handle other implementations)
+
+The are 2 scenario's wrt secret:
+
+- A secret representing the kubeconfig of a cluster. This secret (representing the kubeconfig of the remote cluster) is used to access the cluster. The bootstrap controller install the packages in the staging repo belonging to the cluster when the cluster is ready.
+    - clustername is derive from the annotation: `nephio.org/site`
```

Comment:
We haven't defined the label schema explicitly yet. In https://github.com/nephio-project/nephio-example-packages/blob/main/5g-core-topology/packagevariantset-amf-regional.yaml we picked nephio.org/site-type as being potentially "regional", "edge", "combined". Here we are defining "nephio.org/site". I think we need to standardize on this soon, if not now. Of course, additional labels and implementations may vary, but we should have some sort of standard if we are going to rely upon it to this extent.

I tend to think of site types as things like "global", "regional", "zonal" (maybe), "aggregate", "edge", "far-edge", "near-edge", and that there may be multiple clusters at any given site. So, I think something like:

* nephio.org/site-type - as above
* nephio.org/site - the name or code of a specific site (or should this be nephio.org/site-code?)
* nephio.org/region - calling out one specific site-type code as a separate thing, assuming a site hierarchy
* nephio.org/cluster-type - "workload" or "management"
* nephio.org/cluster-name - for clusters themselves, this is not needed as there is metadata.name, but we could use it for ancillary resources like these secrets

It's not perfect, but should get us pretty far, I think.

Reply:
agreed. I will use nephio.org/cluster-name for this one.